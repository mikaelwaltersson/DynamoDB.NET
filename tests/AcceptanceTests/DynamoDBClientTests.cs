
using Amazon.DynamoDBv2;
using DynamoDB.Net.Model;
using DynamoDB.Net.Serialization;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Options;

namespace DynamoDB.Net.Tests.AcceptanceTests;

public partial class DynamoDBClientTests : IAsyncLifetime
{
    IServiceProvider serviceProvider = null!;
    IAmazonDynamoDB dynamoDB = null!;
    IDynamoDBClient dynamoDBClient = null!;
    string tableName = null!;

    async Task IAsyncLifetime.InitializeAsync()
    {
        var services = new ServiceCollection();

        services.AddOptions();
        services.AddLogging();
        
        services.AddSingleton(
            typeof(IAmazonDynamoDB),
            serviceProvider =>
                new AmazonDynamoDBClient(
                    new AmazonDynamoDBConfig
                    {
                        ServiceURL = "http://localhost:8000"
                    }));

        services.AddSingleton<IDynamoDBClient, DynamoDBClient>();
        services.Configure<DynamoDBClientOptions>(
            options => 
            {
                options.TableNamePrefix = $"acceptance-tests-{Guid.NewGuid()}";
            }
        );

        serviceProvider = services.BuildServiceProvider();

        dynamoDB = serviceProvider.GetRequiredService<IAmazonDynamoDB>();
        
        var createTableRequest = 
            TableDescription.Get(
                typeof(TestModels.UserPost), 
                JsonContractResolver.DefaultDynamoDB).
                    GetCreateTableRequest(
                        serviceProvider.GetRequiredService<IOptions<DynamoDBClientOptions>>().Value);

        await dynamoDB.CreateTableAsync(createTableRequest);

        dynamoDBClient = serviceProvider.GetRequiredService<IDynamoDBClient>();
        tableName = createTableRequest.TableName;
    }

    async Task IAsyncLifetime.DisposeAsync()
    {
        var dynamoDb = serviceProvider.GetRequiredService<IAmazonDynamoDB>();
        
        await dynamoDb.DeleteTableAsync(tableName);
    }
}
