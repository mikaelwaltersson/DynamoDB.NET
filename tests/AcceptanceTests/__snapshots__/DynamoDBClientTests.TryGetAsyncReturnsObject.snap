{
  "UserId": "4793561c-e693-4e58-b31f-99bd9df840ab",
  "Timestamp": "2022-10-18T16:42:00Z",
  "RoleIds": [
    "3d023a08-ed04-4bd5-b9c2-32c11a7b8199",
    "aef0a60f-f5e5-416c-bfa4-9d7c2f66d207"
  ]
}
